float32[3] position
float32 orientation_z
